lib/std/Build/Step/Compile.zig

```diff
@@ -42,7 +42,7 @@ unwind_tables: ?bool,
 compress_debug_sections: enum { none, zlib } = .none,
 lib_paths: ArrayList(LazyPath),
 rpaths: ArrayList(LazyPath),
-framework_dirs: ArrayList(LazyPath),
+framework_dirs: ArrayList(FrameworkDir),
```

Comment:
I think this is a step in the right direction, but should these two union fields, `path` and `path_system`, perhaps be added to `IncludeDir` instead, as `framework_path` and `framework_path_system`?

This could be effectively a non-breaking change despite the `framework_dirs` field being deleted because most users use `addFrameworkPath` and that function could be updated to append to `include_dirs` instead.

Reply:
Mhm, I like that idea, thanks!

Reply:
Done in 71cc2e6